stereo_points/stereo_points.py

```diff
@@ -203,96 +201,79 @@ def onControlPointSelectedChanged(self, newNode):
 
         if slicer.mrmlScene.GetNodesByName(coordTableName).GetNumberOfItems() == 0:
             # print('create new table')
-            coordTable = slicer.vtkMRMLTableNode()
-            coordTable.SetName(coordTableName)
-            coordTable.SetLocked(True)
+            self.coordTableNode = slicer.vtkMRMLTableNode()
+            self.coordTableNode.SetName(coordTableName)
+            self.coordTableNode.SetLocked(True)
             for col in ['Marker', 'x', 'y', 'z', 'r', 'a', 'd', 'X', 'Y', 'Z', 'R', 'A', 'S', 'i', 'j', 'k']:
-                c = coordTable.AddColumn()
+                c = self.coordTableNode.AddColumn()
                 c.SetName(col)
 
-            slicer.mrmlScene.AddNode(coordTable)
-            # keep a map of the fiducialNode <-> TableNode couples
-            self.fiducialNodesToCoordTables_id_tuples.append((newNode.GetID(), coordTable.GetID()))
+            slicer.mrmlScene.AddNode(self.coordTableNode)
+            # refererence the line from the table so we can get back to it
+            self.coordTableNode.AddNodeReferenceID("vtkMRMLMarkupsLineNode", newNode.GetID())
+            newNode.AddNodeReferenceID("vtkMRMLTableNode", self.coordTableNode.GetID())
             # add an observer for both nodes, whenever one is renamed, the other one is renamed as well
-            coordTable.AddObserver(vtk.vtkCommand.ModifiedEvent, self.onCoordTableModified)
+            self.coordTableNode.AddObserver(vtk.vtkCommand.ModifiedEvent, self.onCoordTableModified)
```

Comment:
Need to store the observer tag that this method returns and when switching to a different `self.coordTableNode` then `RemoveObserver` must be called for the previously observed table.

Reply:
Thanks, I've implemented it in [2609bc0](https://github.com/JBeninca/SlicerStereotaxia/pull/9/commits/2609bc03eeb6b2b3209266d4fb710e6355d968e3). I'm not sure if the implementation is correct as I don't really understand why this is needed.